File: Java/백준/Bronze/3052. 나머지/나머지.java
import java.util.*;

public class Main {
    public static void main(String[] args) {

        Scanner s = new Scanner(System.in);

        int[] dividend = new int[10];
        List<Integer> remainder = new ArrayList<>();

        for (int i = 0; i < 10; i++) {
            dividend[i] = s.nextInt();
            remainder.add(dividend[i] % 42);
        }

        Set<Integer> remainderSet = new HashSet<>(remainder);
        System.out.print(remainderSet.size());

        s.close();
    }
}

File: C11/백준/Bronze/3052. 나머지/나머지.c
#include <stdio.h>

int main(void){
	
	int dividend[10];
	int remainder[10];
	
	for(int i = 0; i < 10; i++){
		scanf("%d", &dividend[i]);
		remainder[i] = dividend[i] % 42;
	}
	
	int n = 1;
	int set[10];
	set[0] = remainder[0];
	
	for(int i = 1; i < 10; i++){
		int found = 0;
		for(int j = 0; j < n; j++){
			if(remainder[i] == set[j]){
				found = 1;
				break;
			}
		}
		if(!found){
			set[n] = remainder[i];
			n++;
		}
	}
	
	printf("%d", n);
	
	return 0;
}